compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -462,7 +480,12 @@ fn subst_and_check_impossible_predicates<'tcx>(
 
     let mut predicates = tcx.predicates_of(key.0).instantiate(tcx, key.1).predicates;
     predicates.retain(|predicate| !predicate.needs_subst());
-    let result = impossible_predicates(tcx, predicates);
+
+    let trait_ref = tcx
+        .trait_of_item(key.0)
+        .map(|trait_def_id| ty::TraitRef::from_method(tcx, trait_def_id, key.1));
```

Comment:
Hmm.. could we just construct the trait-ref into a predicate and add it onto the predicates above?

Reply:
How can I do that?  I'm not very literate with predicates.

Reply:
I think just do the logic at https://github.com/rust-lang/rust/pull/91743/files#diff-b81e352311722abf2e93dff8b5af345f3a434adf9c7c068b17de9c815ec114d7R455 

`ty::Binder::dummy(trait_ref).to_poly_trait_predicate().to_predicate(tcx)` 

here here instead of inside of the body of `impossible_predicates`